splice_machine/src/main/java/com/splicemachine/derby/catalog/TriggerNewTransitionRows.java

```diff
@@ -0,0 +1,371 @@
+                        !tec.hasTriggeringResultSet() &&
+                        ConnectionUtil.getCurrentLCC() != lcc &&
+                        lcc.getTriggerExecutionContext() != null) {
+
+	                TriggerExecutionContext currentTEC =
+                            ConnectionUtil.getCurrentLCC().getTriggerExecutionContext();
+                        if (currentTEC != null)
+                            ConnectionUtil.getCurrentLCC().popTriggerExecutionContext(currentTEC);
+                        tec = lcc.getTriggerExecutionContext();
+                        ConnectionUtil.getCurrentLCC().pushTriggerExecutionContext(tec);
+                    }
+                    if (ConnectionUtil.getCurrentLCC().getTriggerExecutionContext() == null)
+                        ConnectionUtil.getCurrentLCC().pushTriggerExecutionContext(tec);
+                }
+	        catch (SQLException e) {
+
+                }
+                if (rowHolder != null) {
+
+                    ConnectionContext cc =
+                    (ConnectionContext) lcc.getContextManager().
+                    getContext(ConnectionContext.CONTEXT_ID);
+                    if (lcc.getTriggerExecutionContext() == null)
+                        lcc.pushTriggerExecutionContext(tec);
+
+                    tec.setConnectionContext(cc);
+                    rowHolder.setActivation(activation);
+                    tec.setTriggeringResultSet(rowHolder.getResultSet());
+                    try {
+                        if (resultSet != null)
+                            resultSet.close();
+                        resultSet = tec.getNewRowSet();
+                    } catch (SQLException e) {
+                        throw Exceptions.parseException(e);
+                    }
+                }
+            }
+        }
+
+	protected ResultSet initializeResultSet() {
+		try {
+                    if (resultSet != null)
+                            resultSet.close();
+
+                    tec = Factory.getTriggerExecutionContext();
+                    if (tec != null)
+                        resultSet = tec.getNewRowSet();
+                }
+		catch (SQLException e) {
+		    // This may happen on initial deserialization.
+                    // Don't crash.  We will fill in the tec later
+                    // in a subsequent deserialization.
+                }
+
+		return resultSet;
+	}
+    
+    public ResultSetMetaData getMetaData() throws SQLException
+    {
+        if (resultSet != null)
+            return resultSet.getMetaData();
+        return null;
+    }
+
+    public void close() throws SQLException {
+       if (resultSet != null) {
+           resultSet.close();
+           resultSet = null;
+       }
+       if (sourceSet != null) {
+           sourceSet.unpersistIt();
+           sourceSet = null;
+       }
+   }
+
+    @Override
+    public double getEstimatedRowCount(VTIEnvironment vtiEnvironment) throws SQLException {
+        return 1000;
```

Comment:
Same, add comment to explain why we use this magic value

Reply:
Added constants and labelled this as a TODO item.  I'll open a Jira for better cost and cardinality estimation.

Reply:
Opened [DB-9103](https://splicemachine.atlassian.net/browse/DB-9103)